test/unit/DominantJuice.unit.t.sol

```diff
@@ -0,0 +1,962 @@
+        vm.warp(3600); // warp 1 hour
+        pledge(pledger1, MIN_PLEDGE_AMOUNT);
+
+        vm.warp(86400); // warp 1 day
+        uint256 hourOfPledge = (block.timestamp - _cycle.start) / 3600;
+        pledge(pledger2, MIN_PLEDGE_AMOUNT * 3);
+
+        vm.warp(604800); // warp 1 week
+        pledge(pledger3, MIN_PLEDGE_AMOUNT);
+
+        DominantJuice.FundingStatus memory status = dominantJuice.getCycleFundingStatus();
+        (uint256 pledger1W, uint256 total1W) = dominantJuice.getPledgerAndTotalWeights(pledger1);
+        (uint256 pledger2W,) = dominantJuice.getPledgerAndTotalWeights(pledger2);
+        (uint256 pledger3W, uint256 total3W) = dominantJuice.getPledgerAndTotalWeights(pledger3);
+
+        assertEq(MIN_PLEDGE_AMOUNT * 5, status.totalPledged);
+        uint256 percent = (100 * status.totalPledged) / CYCLE_TARGET;
+        assertEq(percent, status.percentOfGoal); // Should be 5% (5 * min pledge)
+        assertEq(false, status.isTargetMet);
+
+        uint256 pledger2calculatedW = unwrap(rateOfDecay.powu(hourOfPledge)) * MIN_PLEDGE_AMOUNT * 3;
+        uint256 calculatedTotalWeight = pledger1W + pledger2W + pledger3W;
+
+        assertEq(pledger2calculatedW / 1e18, pledger2W); // Why is this off by a factor of 1e18?
+        assertEq(calculatedTotalWeight, total3W);
+        assertLt(pledger3W, pledger1W); // pledger3W is less since pledged much later
+        assertEq(total1W, total3W); // Getter calls happen after pledges, so total weight doesn't change.
+    }
+
+    function test_didPay_simultaneousPledgersHaveSameWeight() public bonusDeposited {
+        pledge(pledger2, MIN_PLEDGE_AMOUNT);
+        pledge(pledger3, MIN_PLEDGE_AMOUNT);
+        vm.warp(3601);
+        pledge(pledger1, MIN_PLEDGE_AMOUNT);
+
+        (uint256 pledger1W,) = dominantJuice.getPledgerAndTotalWeights(pledger1);
+        (uint256 pledger2W,) = dominantJuice.getPledgerAndTotalWeights(pledger2);
+        (uint256 pledger3W,) = dominantJuice.getPledgerAndTotalWeights(pledger3);
+
+        assertEq(pledger2W, pledger3W);
+        assertLt(pledger1W, pledger2W); // Pledger1 should have less weight since pledged 1 hour later
+    }
+
+    function testFuzz_didPay_simultaneousPledgersDiffAmounts(uint256 _pledger2Amount) public bonusDeposited {
+        vm.warp(10 days);
+        _pledger2Amount = bound(_pledger2Amount, 90 ether, 110 ether);
+        pledge(pledger2, _pledger2Amount);
+        pledge(pledger3, 100 ether);
+
+        (uint256 pledger2W,) = dominantJuice.getPledgerAndTotalWeights(pledger2);
+        (uint256 pledger3W, uint256 totalW) = dominantJuice.getPledgerAndTotalWeights(pledger3);
+
+        if (_pledger2Amount < 100 ether) {
+            assertLt(pledger2W, pledger3W);
+        } else {
+            assertGe(pledger2W, pledger3W);
+        }
+
+        assertEq(pledger2W + pledger3W, totalW); // total weight should always be sum of both weights
+    }
+
+    // More didPay fuzz testing with multiple pledgers
```

Comment:
same pledger, multiple pledges

Reply:
test scenarios with explicit pledge amounts to ensure calculations are correct, and be sure to avoid using any of the calculations from the original contract in your test. Either rewrite the calcs in the test OR better yet come up with the calcs and check against explicit values.

Reply:
Yes sir, all sounds good and makes sense.